File: Dataset/Leetcode/valid/53/584.java
 class Solution {
    public int XXX(int[] nums) {
        int n = nums.length;
        int[] dp = new int[n];
        dp[0] = nums[0];
        int maxSum = nums[0];
        for (int i = 1; i < n; i++) {
            if (dp[i - 1] > 0) {
                dp[i] = dp[i - 1] + nums[i];
            } else {
                dp[i] = nums[i];
            }
            maxSum = Math.max(maxSum, dp[i]);
        }
        return maxSum;
    }
}

File: Dataset/Leetcode/valid/3/584.cpp
 class Solution {
public:
    int XXX(string s) {
        bool arr[128] = {false};
        // unordered_set<char> pre;
        int left = 0;
        int right = 0;
        int length = s.length();
        int max = 0;
        while(right < length){
            while(right < length && !arr[s[right]]){
                arr[s[right]] = true;
                max = ::max(max,right - left + 1);
                ++right;
            }
            if(right < length){
                arr[s[left]] = false;
                left++;
            }
        }
        return max;
    }
};

